### DISCUSSION TITLE
Mock.Of<T>().Get();

### DISCUSSION THREAD

**Discussion by jsk95** [ashmind/SourceMock#6] [Q&A] [answered]
Would like to know the details of the technical challenges of removing `.Get()`

**Comment by ashmind** [comment, marked as answer]
The main one is that you have to have specific types for `Setup` and `Calls`.
Those specific types cannot be returned from a generic, `Of<T>` can only return either `T` or `X<T>`, and there is no way for either to provide a custom surface.

However I found a better idea than `Get` (which might be more reusable too), so I am going to try it out.

It will look like this:
```csharp
[assembly: MockTypesInAssemblyOf(typeof(IParser), ...)]

using ParserNamespace.Mocks;

var parser = new MockIParser();
```
